src/nat/front_ends/fastapi/fastapi_front_end_plugin_worker.py

```
@@ -1088,6 +1090,183 @@ async def redirect_uri(request: Request):
+                    if client is None:
+                        raise RuntimeError(f"MCP client not found for group {group_name}")
+
+                    try:
+                        session_healthy = False
+                        server_tools: dict[str, Any] = {}
+
+                        try:
+                            server_tools = await client.get_tools()
+                            session_healthy = True
+                        except Exception as e:
+                            logger.exception(f"Failed to connect to MCP server {client.server_name}: {e}")
```

Ruff (0.13.1)

Redundant exception object included in `logging.exception` call

(TRY401)